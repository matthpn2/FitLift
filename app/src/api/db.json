{
  "users": {
      "Swagmo996": {
          "name": "Teemo",
          "password": "teemoisgod69"
      },
      "kevanhn": {
          "name": "Kevan",
          "password": "asdfasdf1"
      }
  },
  "new_exercises" : {
    "SAMPLE_USER" : {
      "-L5GDNmo8uZxet9S7mE2" : {
        "reps" : 12,
        "timeStamp" : 1518407604,
        "type" : "Bicep Curls"
      },
      "-L5GDeJnjbOj1SkQRJK8" : {
        "reps" : 5,
        "timeStamp" : 1518407604,
        "type" : "Shoulder Press"
      },
      "-L5GDeeBKCPlTnm-yCiX" : {
        "reps" : 6,
        "timeStamp" : 1518407604,
        "type" : "Lateral Raises"
      }
    }
  },
  "exercises": {
      "Swagmo996": {
          "bicep_curls": {
              "last_set": "set2",
              "sets": {
                  "set1": {
                      "weight": 45,
                      "unit": "lbs",
                      "repetitions": 5,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 45,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  }
              }
          },
          "overhead_press": {
              "last_set": "set4",
              "sets": {
                  "set1": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 8,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 4,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516858963
                  },
                  "set4": {
                      "weight": 70,
                      "unit": "lbs",
                      "repetitions": 3,
                      "timestamp": 1516859963
                  }
              }
          },
          "chest_press": {
              "last_set": "set3",
              "sets": {
                  "set1": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 12,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 11,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 55,
                      "unit": "lbs",
                      "repetitions": 9,
                      "timestamp": 1516858963
                  }
              }
          }
      },
      "kevanhn": {
          "bicep_curls": {
              "last_set": "set3",
              "sets": {
                  "set1": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 5,
                      "timestamp": 1516857363
                  },
                  "set2": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  },
                  "set3": {
                      "weight": 40,
                      "unit": "lbs",
                      "repetitions": 7,
                      "timestamp": 1516857963
                  }
              }
          }
      }
  },
  "friends": {
      "Swagmo996": {
          "kevanhn": true
      },
      "kevanhn": {
          "Swagmo996": true
      }
  }
}