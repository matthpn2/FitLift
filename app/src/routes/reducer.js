import { NavigationActions } from 'react-navigation';
import { Navigation } from './index';

const initialState = Navigation.router.getStateForAction(
  Navigation.router.getActionForPathAndParams('login')
);

export default (state = initialState, action) => {
  switch (action.type) {
    case 'LOGIN_SUCCESS':
    case 'CREATE_USER_SUCCESS':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({ routeName: 'loggedIn' })
      );
    case 'logout':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({ routeName: 'login' })
      );
    case 'DISPLAY_EXERCISE_DAY':
      return Navigation.router.getStateForAction(
        NavigationActions.navigate({
          params: {
            day: action.day
          },
          routeName: 'exerciseDay'
        }),
        Navigation.router.getStateForAction(action, state)
      );
    default:
      return Navigation.router.getStateForAction(action, state);
  }
};
