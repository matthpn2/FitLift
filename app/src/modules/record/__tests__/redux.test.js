import reducer, {
  TypeKeys,
  updateNewExercise,
  exercisesToRecordSelector
} from '../redux';
import { new_exercises as allNewExercises } from '../../../api/db.json';

const newExercises = allNewExercises.SAMPLE_USER;

describe('selector tests', () => {
  it('it should work', () => {
    expect(
      exercisesToRecordSelector({
        db: {
          newExercises
        },
        record: {
          modifiedExercises: {
            '-L5GDNmo8uZxet9S7mE2': {
              reps: 12,
              weight: 35
            },
            '-L5GDeJnjbOj1SkQRJK8': {
              reps: 5,
              weight: 15
            },
            '-L5GDeeBKCPlTnm-yCiX': {
              reps: 6,
              weight: 45
            }
          }
        }
      })
    ).toMatchSnapshot();
  });
});

describe('reducer tests', () => {
  it('TypeKeys.LOGIN should work', () => {
    expect(
      reducer(
        {
          isLoading: false
        },
        {
          type: TypeKeys.LOGIN
        }
      )
    ).toEqual({
      isLoading: true
    });
  });

  it('TypeKeys.RECEIVE_NEW_EXERCISES should work', () => {
    expect(
      reducer(
        {
          isLoading: true,
          modifiedExercises: {}
        },
        {
          data: newExercises,
          type: TypeKeys.RECEIVE_NEW_EXERCISES
        }
      )
    ).toEqual({
      isLoading: false,
      modifiedExercises: {
        '-L5GDNmo8uZxet9S7mE2': {
          reps: 12
        },
        '-L5GDeJnjbOj1SkQRJK8': {
          reps: 5
        },
        '-L5GDeeBKCPlTnm-yCiX': {
          reps: 6
        }
      }
    });
  });

  it('TypeKeys.UPDATE_NEW_EXERCISE should work', () => {
    expect(
      reducer(
        {
          isLoading: false,
          modifiedExercises: {
            '-1': {
              reps: 12,
              timeStamp: 1518407604,
              type: 'Bicep Curls',
              weight: 35
            },
            '-2': {
              reps: 5,
              timeStamp: 1518407604,
              type: 'Shoulder Press',
              weight: 85
            }
          }
        },
        updateNewExercise('-1', 'weight', 25)
      )
    ).toMatchSnapshot();
  });

  it('TypeKeys.UPDATE_NEW_EXERCISE should work', () => {
    expect(
      reducer(
        {
          isLoading: false,
          modifiedExercises: {
            '-1': {
              reps: 12,
              timeStamp: 1518407604,
              type: 'Bicep Curls',
              weight: 35
            },
            '-2': {
              reps: 5,
              timeStamp: 1518407604,
              type: 'Shoulder Press',
              weight: 85
            }
          }
        },
        updateNewExercise('-2', 'reps', 25)
      )
    ).toMatchSnapshot();
  });
});
