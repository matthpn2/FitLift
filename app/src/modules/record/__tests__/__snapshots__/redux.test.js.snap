// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer tests TypeKeys.UPDATE_NEW_EXERCISE should work 1`] = `
Object {
  "isLoading": false,
  "modifiedExercises": Object {
    "-1": Object {
      "reps": 12,
      "timeStamp": 1518407604,
      "type": "Bicep Curls",
      "weight": 25,
    },
    "-2": Object {
      "reps": 5,
      "timeStamp": 1518407604,
      "type": "Shoulder Press",
      "weight": 85,
    },
  },
}
`;

exports[`reducer tests TypeKeys.UPDATE_NEW_EXERCISE should work 2`] = `
Object {
  "isLoading": false,
  "modifiedExercises": Object {
    "-1": Object {
      "reps": 12,
      "timeStamp": 1518407604,
      "type": "Bicep Curls",
      "weight": 35,
    },
    "-2": Object {
      "reps": 25,
      "timeStamp": 1518407604,
      "type": "Shoulder Press",
      "weight": 85,
    },
  },
}
`;

exports[`selector tests it should work 1`] = `
Array [
  Object {
    "id": "-L5GDNmo8uZxet9S7mE2",
    "reps": 12,
    "timeStamp": 1518407604,
    "type": "Bicep Curls",
    "weight": 35,
  },
  Object {
    "id": "-L5GDeJnjbOj1SkQRJK8",
    "reps": 5,
    "timeStamp": 1518407604,
    "type": "Shoulder Press",
    "weight": 15,
  },
  Object {
    "id": "-L5GDeeBKCPlTnm-yCiX",
    "reps": 6,
    "timeStamp": 1518407604,
    "type": "Lateral Raises",
    "weight": 45,
  },
]
`;
